src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -96,19 +96,8 @@ public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $q
             $field = $property;
 
             if ($this->isPropertyNested($property)) {
-                $propertyParts = $this->splitPropertyParts($property);
-
-                $parentAlias = $alias;
-
-                foreach ($propertyParts['associations'] as $association) {
-                    $alias = $queryNameGenerator->generateJoinAlias($association);
-                    $queryBuilder->join(sprintf('%s.%s', $parentAlias, $association), $alias);
-                    $parentAlias = $alias;
-                }
-
-                $field = $propertyParts['field'];
-
-                $metadata = $this->getNestedMetadata($resourceClass, $propertyParts['associations']);
+                list($alias, $field, $associations) = $this->addJoinsForNestedProperty($property, $alias, $queryBuilder, $queryNameGenerator);
```

Comment:
The splat operator?

My preference go to returning an array with explicit keys and use the keys in the calling method.